src/openforms/payments/contrib/worldline/plugin.py

```diff
@@ -0,0 +1,255 @@
+
+    def start_payment(
+        self,
+        request: HttpRequest,
+        payment: SubmissionPayment,
+        options: PaymentOptions,
+    ) -> PaymentInfo:
+        order = _generate_order(payment)
+        checkout_input = _generate_checkout_input(request, payment, self)
+        client = options["merchant"].get_checkout_client()
+
+        # See the hostedCheckoutSpecificInput field on
+        # https://docs.direct.worldline-solutions.com/en/api-reference#tag/HostedCheckout/operation/CreateHostedCheckoutApi
+        # for more configurable checkout options.
+        checkout_request = CreateHostedCheckoutRequest()
+        checkout_request.from_dictionary(
+            {
+                "returnCancelState": True,
+                "hostedCheckoutSpecificInput": checkout_input,
+                "order": order,
+            }
+        )
+
+        try:
+            checkout_response = client.create_hosted_checkout(checkout_request)
+        except (ApiException, CommunicationException) as e:
+            payment.status = PaymentStatus.failed
+            payment.save(update_fields=("status",))
+
+            raise APIException(
+                detail="Failed to retrieve redirect URL from payment provider"
+            ) from e
+
+        plugin_options = payment.plugin_options
+        payment.plugin_options = {
+            **plugin_options,
+            "returnmac": checkout_response.returnmac,
+            "checkout_id": checkout_response.hosted_checkout_id,
+        }
+
+        payment.save(update_fields=("plugin_options",))
+
+        return PaymentInfo(
+            type=PaymentRequestType.get,
+            url=checkout_response.redirect_url or "",  # valid for three hours
+            data={},
+        )
+
+    def apply_status(
+        self,
+        payment: SubmissionPayment,
+        worldline_status: str,
+        checkout_status: str,
+        payment_id: str,
+    ) -> None:
+        if payment.status in PAYMENT_STATUS_FINAL:
+            # shouldn't happen or race-condition
+            return
+
+        status = get_payment_status(worldline_status, checkout_status=checkout_status)
+
+        # run this query as atomic update()
+        qs = SubmissionPayment.objects.filter(id=payment.id)
+        qs = qs.exclude(status__in=PAYMENT_STATUS_FINAL)
+        qs = qs.exclude(status=status)
+        res = qs.update(status=status, provider_payment_id=payment_id)
+
+        if res > 0:
+            payment.refresh_from_db()
+
+    def handle_return(
+        self,
+        request: Request,
+        payment: SubmissionPayment,
+        options: PaymentOptions,
+    ) -> HttpResponse:
+        with structlog.contextvars.bound_contextvars(
+            submission_uuid=str(payment.submission.uuid),
+            payment_uuid=str(payment.uuid),
+            plugin=self,
+            entrypoint="browser",
+        ):
+            log = logger.bind()
+            log.info("process_payment_status")
```

Comment:
```suggestion
            logger.info("process_payment_status")
```
the `bind` doesn't do anything